packages/shared/src/components/squads/SquadHeaderBar.spec.tsx

```diff
@@ -0,0 +1,73 @@
+import { QueryClient } from '@tanstack/react-query';
+import { render, RenderResult, screen } from '@testing-library/react';
+import React from 'react';
+import { SquadHeaderBar } from './SquadHeaderBar';
+
+import {
+  generateMembersList,
+  generateTestSquad,
+} from '../../../__tests__/fixture/squads';
+import { TestBootProvider } from '../../../__tests__/helpers/boot';
+
+const client = new QueryClient();
+const mock = {
+  squad: generateTestSquad(),
+  members: generateMembersList().map((member) => member.node),
+} as const;
+
+const renderComponent = (options?: {
+  props?: Partial<typeof mock>;
+}): RenderResult => {
+  const { props = {} } = options ?? {};
+
+  return render(
+    <TestBootProvider client={client}>
+      <SquadHeaderBar
+        members={props.members ?? mock.members}
+        squad={props.squad ?? mock.squad}
+      />
+    </TestBootProvider>,
+  );
+};
+
+describe('Member list', () => {
+  it('should render the squad header bar with the correct number of members', async () => {
+    renderComponent();
+    const { membersCount } = mock.squad;
+    const countEl = await screen.findByLabelText(
+      `View ${membersCount} squad members`,
+    );
+    expect(countEl).toHaveTextContent(membersCount.toString());
+  });
+});
+
+describe('Moderation button', () => {
+  it('should not render button with pending posts moderation is disabled', async () => {
+    const squad = generateTestSquad({
+      moderationRequired: false,
+      moderationPostCount: 10,
+    });
+    renderComponent({ props: { squad } });
+    const buttonEl = screen.queryByTestId('squad-moderation-button');
+    expect(buttonEl).not.toBeInTheDocument();
+  });
+
+  it('should not render button with pending posts if none', async () => {
+    const squad = generateTestSquad({
+      moderationRequired: true,
+      moderationPostCount: 0,
+    });
+    renderComponent({ props: { squad } });
+    const buttonEl = screen.queryByTestId('squad-moderation-button');
```

Comment:
Non blocking but these don't have a visual label?

Reply:
[I used label for query](https://github.com/dailydotdev/apps/pull/3659/files#diff-a3264bb46cdf89b426ae1555fe8fb8f12424649b375bbd23f6864999e1b14cb9R71) when expecting the component to be visible, otherwise used the testid for a quick approach 👀